core/distributed/vector.cpp

```diff
@@ -573,19 +573,62 @@ void Vector<ValueType>::compute_squared_norm2(ptr_param<LinOp> result,
 }
 
 
+template <typename ValueType>
+void Vector<ValueType>::compute_mean(ptr_param<LinOp> result) const
+{
+    array<char> tmp{this->get_executor()};
+    this->compute_mean(result, tmp);
+}
+
+
+template <typename ValueType>
+void Vector<ValueType>::compute_mean(ptr_param<LinOp> result,
+                                     array<char>& tmp) const
+{
+    using MeanVector = local_vector_type;
+    const auto global_size = this->get_size()[0];
+    const auto local_size = this->get_local_vector()->get_size()[0];
+    const auto num_vecs = static_cast<int>(this->get_size()[1]);
+    GKO_ASSERT_EQUAL_COLS(result, dim<2>(1, num_vecs));
```

Comment:
```suggestion
    GKO_ASSERT_EQUAL_COLS(result, this);
```
a few characters shorter